segmentstore/server/src/main/java/io/pravega/segmentstore/server/logs/ContainerMetadataUpdateTransaction.java

```diff
@@ -505,11 +501,12 @@ private long generateUniqueSegmentId() {
         // The ContainerMetadata.SequenceNumber is always guaranteed to be unique (it's monotonically strict increasing).
         // It can be safely used as a new unique Segment Id. If any clashes occur, just keep searching up until we find
         // a non-used one.
-        long segmentId = Math.max(this.baseMetadata.getOperationSequenceNumber(), ContainerMetadata.NO_STREAM_SEGMENT_ID + 1);
+        long segmentId = this.realMetadata.getOperationSequenceNumber();
```

Comment:
When we assign segment ids, I want them to be unique. I can either generate a random number and hope it's unique, but since not all segments are in memory, it's impossible to do it that way. Another thing that is guaranteed to be unique is the Operation Sequence Number. It is a monotonically strict-increasing number, which changes with every new operation. Since the only way to assign a SegmentId to a Segment is via an Operation, this guarantees uniqueness.  The subsequent while loop is just a fail-safe in case we do end up with some sort of duplicate.